### PR TITLE
[Agency Dashboard] Create internal home page for Agency Dashboards

### LINKED ISSUE #597: [Dashboards] Create table of contents for Agency Dashboards
Right now, you have to know the agency ID in order to navigate to a particular agency's dashboard (e.g. https://justice-counts-agency-dashboard-web-qqec6jbn6a-uc.a.run.app/agency/147). This makes it really hard to test things out!

It'd be nice to have a "homepage" (fine if this is ugly, since it would be internal only) at https://justice-counts-agency-dashboard-web-qqec6jbn6a-uc.a.run.app that links out to all the different agencies, to make it easy to flip through them and test out how the dashboard looks on different kinds of data.

To do this, we'd need a new backend endpoint (without auth_decorator, since this is for the public dashboard) that returns a JSON structure sort of like this:

```
"agencies": [
   { "name": XXX, "id": XXX},
   { "name": XXX, "id": XXX},
]
```

We should be able to use the backend method `get_agencies_by_id` to accomplish this.

The frontend could then use this to render a list of URLs.

An extension of this task, to make this homepage even more useful potentially, would be to indicate next to each agency's name whether it has any published data, and if so, for how many metrics. To do this, we'd need to extend the backend endpoint to look more like this:

```
"agencies": [
   { "name": XXX, "id": XXX, "number_of_published_metrics": XXX},
]
```

To do this, we could re-use some of the logic from the `get_agency_published_data` endpoint. We won't need to re-use all of it, because we don't actually need to return all the published datapoints -- we just need to know whether any exist and if so how many.

### PR DESCRIPTION
**Pull request by mxosman**
## Description of the change

Creates an internal home page for Agency Dashboards that shows a list of all agencies (w/ a link to each agency's dashboard) and their # of published metrics.

I also took this opportunity to reorganize some of the files, exports, imports (happy to separate this out into another PR if it feels like it's polluting this one).

Here are the main files to focus on:
* `src/stores/AgencyDataStore.ts`
* `src/Home/Home.tsx`
* `src/Home/Home.styles.tsx`
* `src/Home/types.ts`
* `src/App.tsx`

Demo: https://mahmoudtest---justice-counts-agency-dashboard-web-qqec6jbn6a-uc.a.run.app/

Backend companion PR: https://github.com/Recidiviz/recidiviz-data/pull/20710

## Related issues

Closes #597

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: agency-dashboard/src/Home/Home.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    <>
+      <Styled.HomeContainer>
+        <HeaderBar />
+        <Styled.Title>Welcome to Agency Dashboards</Styled.Title>
+
+        {agencyDataStore.loading ? (
+          <Loader />
+        ) : (
+          <Styled.AgencyDetailsContainer>
+            {agenciesMetadata.map((agency) => (
```
Comment (lilidworkin) [inline agency-dashboard/src/Home/Home.tsx:53]: Such a nit, but can sort the agencies alphabetically?

**Review by lilidworkin — APPROVED** [review]
OMG -- you made this look SO PRETTY. I'm obsessed!

**Comment by lilidworkin** [comment]
Another super nit -- I'm noticing that there is sometimes a bit of a lag when you navigate back and forth between agencies, the old agency still shows up for a second. Is it possible to replace that with a spinner? Here's an example:

https://user-images.githubusercontent.com/99899591/236476899-e6ee2b62-7f3e-4a1a-be60-9ef58d0bd89f.mov

This is SO not a bit deal but if it's easy to fix, might as well!

File: agency-dashboard/src/Home/Home.tsx (hunk 1 above)
Comment (mxosman) [reply agency-dashboard/src/Home/Home.tsx:53]: Absolutely!

**Comment by mxosman** [comment]
> Another super nit -- I'm noticing that there is sometimes a bit of a lag when you navigate back and forth between agencies, the old agency still shows up for a second. Is it possible to replace that with a spinner? Here's an example:
> 
> This is SO not a bit deal but if it's easy to fix, might as well!

Great call out - saw the same thing too but haven't pursued it yet! Will take a look - great idea re: the spinner - let me see why it's doing that in the first place.

**Comment by mxosman** [comment]
OK - everything should be updated and live in the testing URL (https://mahmoudtest---justice-counts-agency-dashboard-web-qqec6jbn6a-uc.a.run.app/) - agencies are sorted by name and a loader shows up while we're fetching new agency data. Thank you so much for these suggestions!

**Review by lilidworkin — APPROVED** [review]
Looks perfect! I also just showed this to Juan and he was like OMG I love this I'm bookmarking it right now haha :D

**Comment by morden35** [comment]
I'm loving this

**Pull request merged**